server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -173,14 +181,39 @@ public void define(Service service) {
         @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
     })
     private TaskIdDto userActions(Request request, Response response) throws JsonExtractException {
-        UserVaultAction requestedAction = extractUserVaultAction(request);
+        VaultAction requestedAction = extractVaultAction(request);
 
         Task requestedTask = generateTask(requestedAction, request);
         TaskId taskId = taskManager.submit(requestedTask);
         return TaskIdDto.respond(response, taskId);
     }
 
-    private Task generateTask(UserVaultAction requestedAction, Request request) throws JsonExtractException {
+    @POST
+    @Path(ROOT_PATH)
+    @ApiOperation(value = "Purge all expired messages base on retentionPeriod of deletedMessageVault configuration")
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            required = true,
+            name = "action",
+            dataType = "String",
+            paramType = "query",
+            example = "?action=purge",
+            value = "Compulsory. Needs to be a purge action")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.CREATED_201, message = "Task is created", response = TaskIdDto.class),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Bad request - action is invalid"),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
+    })
+    private TaskIdDto purgeActions(Request request, Response response) {
```

Comment:
That route seems quite specific to the purge operation and will probably never do anything else, so why would we need an implicit param `action=purge` ?

Reply:
the route path is : `http://ip:port/deletedMessages?action=purge`. I think it's is more clear to put the action param.

Reply:
Then that seems like a more general route, I'm not sure we can call the method `purgeActions` then. Wdyt?

Reply:
ah, now I get what you meant, it can be re named to `globalActions` IMHO

Reply:
Not sure if it's the best name we can come up with, but that would be better yes !